src/sentry/lang/native/applecrashreport.py

```diff
@@ -0,0 +1,126 @@
+        return "OS Version: {} {} ({})\nReport Version: {}".format(
+            self.context.get('os').get('name'),
+            self.context.get('os').get('version'),
+            self.context.get('os').get('build'),
+            REPORT_VERSION
+        )
+
+    def get_threads_apple_string(self):
+        rv = []
+        for thread in self.threads:
+            thread_string = self.get_thread_apple_string(thread)
+            if thread_string is not None:
+                rv.append(thread_string)
+        return "\n\n".join(rv)
+
+    def get_thread_apple_string(self, thread):
+        rv = []
+        stacktrace = thread.get('stacktrace')
+        if stacktrace is None:
+            return None
+        if stacktrace:
+            frames = stacktrace.get('frames')
+            if frames:
+                i = 0
+                for frame in reversed(frames):
+                    frame_string = self._convert_frame_to_apple_string(
+                        frame=frame,
+                        number=i
+                    )
+                    if frame_string is not None:
+                        rv.append(frame_string)
+                        i += 1
+
+        if len(rv) == 0:
+            return None  # No frames in thread, so we remove thread
+        thread_string = 'Thread {} name: {}\n'.format(thread['id'],
+            thread['name'] and thread['name'] or ''
+        )
+        if thread['crashed']:
+            thread_string += 'Thread {} Crashed:\n'.format(thread['id'])
+        return thread_string + "\n".join(rv)
+
+    def _convert_frame_to_apple_string(self, frame, number=0):
+        if frame.get('instruction_addr') is None:
+            return None
+        slide_value = self._get_slide_value(frame['image_addr'])
+        instruction_addr = slide_value + int(frame['instruction_addr'], 16)
+        image_addr = slide_value + int(frame['image_addr'], 16)
+        offset = ''
+        if frame['image_addr'] is not None and not self.symbolicated:
+            offset = ' + {}'.format(
+                instruction_addr - slide_value - int(frame['symbol_addr'], 16)
+            )
+        symbol = hex(image_addr)
+        if self.symbolicated:
+            file = ''
+            if frame.get('filename') and frame.get('lineno'):
+                file = " ({}:{})".format(
+                    frame['filename'],
+                    frame['lineno']
+                )
+            symbol = "{}{}".format(
+                frame['function'],
+                file
+            )
+        return "{}{}{}{}{}".format(
+            str(number).ljust(4, " "),
+            frame['package'].rsplit('/', 1)[-1].ljust(32, " "),
+            hex(instruction_addr).ljust(20, " "),
+            symbol,
+            offset
+        )
+
+    def _get_slide_value(self, image_addr):
+        if self.debug_images:
+            for debug_image in self.debug_images:
+                if int(debug_image['image_addr'], 16) == int(image_addr, 16):
```

Comment:
This should use `parse_addr` since there is no guarnatee currently which format those things are I think.